Rivet/Functions/Update-Database.ps1

```diff
@@ -133,38 +131,31 @@ function Update-Database
         $byName['Include'] = $Name
     }
 
-    Get-MigrationFile -Path $Path -Configuration $Configuration @byName -ErrorAction Stop |
+    $query = 'if (object_id(''{0}'', ''U'') is not null) select ID, Name, Who, AtUtc from {0}' -f $RivetMigrationsTableFullName
+    $appliedMigrations = @{}
+    foreach( $migration in (Invoke-Query -Query $query) )
+    {
+        $appliedMigrations[$migration.ID] = $migration
+    }
+
+    $migrations = 
+        Get-MigrationFile -Path $Path -Configuration $Configuration @byName -ErrorAction Stop |
         Sort-Object -Property 'MigrationID' -Descending:$popping |
         Where-Object {
-
             if( $RivetSchema )
             {
                 return $true
             }
 
             if( [int64]$_.MigrationID -lt 1000000000000 )
             {
-                Write-Error ('Migration ''{0}'' has an invalid ID. IDs lower than 01000000000000 are reserved for internal use.' -f $_.FullName)
-                $stopMigrating = $true
+                Write-Error ('Migration "{0}" has an invalid ID. IDs lower than 01000000000000 are reserved for internal use.' -f $_.FullName) -ErrorAction Stop

```

Comment:
return after -ErrorAction Stop.

Reply:
That's a convention we use to it's clear that the code is returning.